types/backbone/index.esm.d.ts

```diff
@@ -0,0 +1,595 @@
+    model?: TModel | undefined;
+    // TODO: quickfix, this can't be fixed easy. The collection does not need to have the same model as the parent view.
+    collection?: Collection<any> | undefined; // was: Collection<TModel>;
+    el?: TElement | JQuery | string | undefined;
+    id?: string | undefined;
+    attributes?: Record<string, any> | undefined;
+    className?: string | undefined;
+    tagName?: string | undefined;
+    events?: _Result<EventsHash> | undefined;
+}
+
+export type ViewEventListener = (event: JQuery.Event) => void;
+
+export class View<TModel extends (Model | undefined) = Model, TElement extends Element = HTMLElement>
+    extends EventsMixin
+    implements Events
+{
+    /**
+     * Do not use, prefer TypeScript's extend functionality.
+     */
+    static extend(properties: any, classProperties?: any): any;
+
+    /**
+     * For use with views as ES classes. If you define a preinitialize
+     * method, it will be invoked when the view is first created, before any
+     * instantiation logic is run.
+     * @see https://backbonejs.org/#View-preinitialize
+     */
+    preinitialize(options?: ViewOptions<TModel, TElement>): void;
+
+    constructor(options?: ViewOptions<TModel, TElement>);
+    initialize(options?: ViewOptions<TModel, TElement>): void;
+
+    /**
+     * Events hash or a method returning the events hash that maps events/selectors to methods on your View.
+     * For assigning events as object hash, do it like this: this.events = <any>{ "event:selector": callback, ... };
+     * That works only if you set it in the constructor or the initialize method.
+     */
+    events: _Result<EventsHash>;
+
+    // A conditional type used here to prevent `TS2532: Object is possibly 'undefined'`
+    model: TModel extends Model ? TModel : undefined;
+    collection: Collection<any>;
+    setElement(element: TElement | JQuery): this;
+    id?: string | undefined;
+    cid: string;
+    className?: string | undefined;
+    tagName: string;
+
+    el: TElement;
+    $el: JQuery;
+    attributes: Record<string, any>;
+    $(selector: string): JQuery;
+    render(): this;
+    remove(): this;
+    delegateEvents(events?: _Result<EventsHash>): this;
+    delegate(eventName: string, selector: string, listener: ViewEventListener): this;
+    undelegateEvents(): this;
+    undelegate(eventName: string, selector?: string, listener?: ViewEventListener): this;
+
+    protected _removeElement(): void;
+    protected _setElement(el: TElement | JQuery): void;
+    protected _createElement(tagName: string): void;
+    protected _ensureElement(): void;
+    protected _setAttributes(attributes: Record<string, any>): void;
+}
+
+// SYNC
+export function sync(method: string, model: Model | Collection, options?: JQueryAjaxSettings): any;
+export function ajax(options?: JQueryAjaxSettings): JQueryXHR;
+export let emulateHTTP: boolean;
+export let emulateJSON: boolean;
+
+// Utility
+export function noConflict(): any;
```

Comment:
This is a loss of information compared to the UMD version. This function returns an object of the same type as the default export.